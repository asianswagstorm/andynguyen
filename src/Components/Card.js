import React, { Component } from 'react';

class Card extends Component {
  
    render = () => {
        return (
            <div className="media">
                <a target = {this.props.linkType === "external" ? "_blank" : "_self"} rel="noopener noreferrer" href={this.props.href}><img src={this.props.imgSrc} alt="" title={this.props.href} /> </a>
            </div>
        );
    };
};

export default Card;
