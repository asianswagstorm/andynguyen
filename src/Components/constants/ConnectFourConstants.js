/**
 * column x row 
 * //An array of linked lists ???  
 */
export const connectFourBoard = [
    {columnNumber : 0 , rowArrays: ["","","","","",""]}, 
    {columnNumber : 1 , rowArrays: ["","","","","",""]}, 
    {columnNumber : 2 , rowArrays: ["","","","","",""]}, 
    {columnNumber : 3 , rowArrays: ["","","","","",""]}, 
    {columnNumber : 4 , rowArrays: ["","","","","",""]}, 
    {columnNumber : 5 , rowArrays: ["","","","","",""]}, 
    {columnNumber : 6 , rowArrays: ["","","","","",""]}
];
   

//[column,row] 
export const boardIndex = [
    [[0,0], [1,0], [2,0], [3,0] , [4,0], [5,0], [6,0]],
    [[0,1], [1,1], [2,1], [3,1] , [4,1], [5,1], [6,1]],
    [[0,2], [1,2], [2,2], [3,2] , [2,4], [5,2], [6,2]],
    [[0,3], [1,3], [2,3], [3,3] , [3,4], [5,2], [6,3]],
    [[0,4], [1,4], [2,4], [3,4] , [4,4], [5,2], [6,4]],
    [[0,5], [1,5], [2,5], [3,5] , [5,4], [5,2], [6,5]],
];

export const rowIndexByColumn = [5,5,5,5,5,5,5];
/**
 * unshift
 * column 0 :[]
 * column 1 :[]
 * column 2 :[]
 * column 3 :[]
 * column 4 :[]
 * column 5 :[]
 * column 6 :[]
 */

 //column 0 :["","","","","",""] , pop then push