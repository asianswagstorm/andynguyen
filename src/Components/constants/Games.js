export const TicTacToeWinCombo = array => [   
    [array[0][0], array[0][1], array[0][2]], //row 1 horizontal
    [array[1][0], array[1][1], array[1][2]], //row 2 horizontal
    [array[2][0], array[2][1], array[2][2]], //row 3 horizontal
    [array[0][0], array[1][0], array[2][0]], //col 1 vertical
    [array[0][1], array[1][1], array[2][1]], //col 2 vertical
    [array[0][2], array[1][2], array[2][2]], //col 3 vertical
    [array[0][0], array[1][1], array[2][2]], //col 1 diagonal
    [array[0][2], array[1][1], array[2][0]] //col 3 diagonal
]; 

export const winIndexs = [
    {index1: [0,0], index2: [0,1], index3: [0,2]},
    {index1: [1,0], index2: [1,1], index3: [1,2]},
    {index1: [2,0], index2: [2,1], index3: [2,2]},
    {index1: [0,0], index2: [1,0], index3: [2,]},
    {index1: [0,1], index2: [1,1], index3: [2,1]},
    {index1: [0,2], index2: [1,2], index3: [2,2]},
    {index1: [0,0], index2: [1,1], index3: [2,2]},
    {index1: [0,2], index2: [1,1], index3: [2,0]}
];

export const tictactoe_boxes = [ //overwritten
    ['','',''],
    ['','',''],
    ['','','']
];

export const tictactoe_score_label = [  { label: 'X', id: "user-label"},
                                        {    label: 'O', id: "computer-label"}];

export const playerSymbols = { userLabel : 'X', computerLabel : 'O' };

export const playerPicker = [
    {isComp : false, name: "Human", id: "H-opponent"},
    {isComp : true, name: "Computer", id: "C-opponent"}
];