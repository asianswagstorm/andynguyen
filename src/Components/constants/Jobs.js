import azure from "../../photos/job/azure.png";
import couch from "../../photos/job/couch.png";
import django from "../../photos/job/django.png";
import docker from "../../photos/job/docker.png";
import react from "../../photos/job/react.png";
import redux from "../../photos/job/redux.png";


export const metso = {

job_overview: { job_detail: `I worked at  Metso Mineral in Lachine Quebec for a period of 7 months between July 2019 to January 2020, assisting the Field Service Department to turn a concept idea into a working application. The application allowed the company to manage and track their many client's mill information during mill inspection visits.
                             My role was to convert the initial static HTML code into dynamic functioning application using modern web framework like React and Django. Working directly with the management of the Field Service Department, I worked in an Agile environment completing weekly sprints 
                             in a timely manner.`,
                product_explanation:`The application built is a progressive web application that can run offline.
                                     The main purpose of the application was to change from traditional PDF written reports into a web based format to make it easier to generate mill inspection reports.`
                                    ,
        

                tech_stack: [
                    {
                        name: "Django",
                        image: django,
                        description: "Used the Django framework on the backend.",
                        link: "https://docs.djangoproject.com/en/3.0/" 
                    }, 
                    {
                        name: "React",
                        image: react,
                        description: "Used React framework to build the UI.",
                        link: "https://reactjs.org/docs/getting-started.html" 
                    },
                    {
                        name: "Redux",
                        image: redux,
                        description: "Used Redux as the state manager.",
                        link: "https://redux.js.org/introduction/getting-started" 
                    },
                    {
                        name: "CouchDB / PouchDB",
                        image: couch,
                        description: "Used CouchDB as the database, PouchDB as an indexed DB (offline mode).",
                        link: "https://docs.couchdb.org/en/stable/" 
                    },
                    {
                        name: "Docker",
                        image: docker,
                        description: "Used Docker to virtualize the frontend and backend into docker images.",
                        link: "https://docs.docker.com/" 
                    },
                    {
                        name: "Azure",
                        image: azure,
                        description: `Used Microsoft Azure services to manage project and hosting service.`,
                        link: "https://docs.microsoft.com/en-us/azure/" 
                    }
                ]
                
            },
                /*
                "The Technology stack used was React JS / Redux on the frontend,
                 Python Django on the Backend, CouchDB / PouchDB as the database, Docker as a container, and Azure App Services as the Cloud host server."     
                */
                
// add logo of tech stack??? 

                tasks_performed: [
    {
        title:"Front End Role:",
        detail:"Designed and implemented the web app using React JS and Redux. Used REST services to request API responses from the backend server. "
    },
    {
        title:"Back End Role:",
        detail:"Using Python Django, I wrote API routes that returned HTTP Responses in JSON format to be used in the frontend server. Wrote CouchDB views to filter information based on the documents on the databse."
    },
    {
        title:"DevOps Role:",
        detail:`Dockerized the project into docker images, pushed the images into Azure Registry Containers.
                Deployed the Front End, Back End and the database registry containers on Azure App Services.`
    },
    {
        title:"Team Lead Role:",
        detail:`Taught my colleague how to code in React. Organize daily Scrum meetings. Organized the weekly sprints.`
    }
]

};