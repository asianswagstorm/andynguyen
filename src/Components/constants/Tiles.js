// import { Modal } from "antd";

export const tiles = [
    {
      linkType: "link",
      href: "Jamdo",
      src: "images/thumbs/thumb1.jpg",
      alt: "",
      title:
        "Jamdo, a web application created for Web Services and Applications class.",
      type: ["Python"]
    },
    {
      linkType: "link",
      href: "https://airbnbmock.herokuapp.com/",
      src: "images/thumbs/thumb3.jpg",
      alt: "",
      title:
        "An airbnb replica front page created with react JS (first time using React) ",
      type: ["JavaScript"]
    },
    {
      linkType: "link",
      href: "https://recipes-api-app.herokuapp.com/",
      src: "images/thumbs/thumb12.jpg",
      alt: "",
      title: "A recipe api app made with react. ",
      type: ["JavaScript"]
    },
    {
      linkType: "link",
      href: "https://pokemon-dex-application.herokuapp.com/",
      src: "images/thumbs/thumb13.jpg",
      alt: "",
      title: "A pokedex api app made with react. ",
      type: ["JavaScript"]
    },
    {
      linkType: "link",
      href: "https://spotify-app-andy.herokuapp.com/",
      src: "images/thumbs/thumb14.jpg",
      alt: "",
      title: "A spotify api app made with react. ",
      type: ["JavaScript"]
    },
    {
      linkType: "link",
      href: "/TicTacToe",
      src: "images/thumbs/thumb6.jpg",
      alt: "",
      title: "Tic Tac Toe Game .",
      type: ["JavaScript"]
    },
    {
      linkType: "link",
      href: "/Connect4",
      src: "images/thumbs/thumb8.jpg",
      alt: "",
      title: "Connect 4 Game.",
      type: ["JavaScript"]
    },
    {
      linkType: "modal",
      href: "#",
      src: "images/thumbs/thumb7.jpg",
      alt: "",
      title: "Work Experience",
      type: ["Work Experience", "JavaScript", "Python"]
    }
   
  ];