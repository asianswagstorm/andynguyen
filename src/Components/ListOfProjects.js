import React from 'react';
import {tiles} from "./constants/Tiles";
import Card from "./Card";

const ListOfProjects = (props) => (
    <div className="content">
      {tiles.map((value, index) => 
        (
          value.type.includes(props.filterOption) ||
          props.filterOption === ""
        ) ?(
            <Card
              key={index}
              href={value.href}
              imgSrc={value.src}
              linkType = {value.linkType}
              alt=""
              title={value.title}
            />) : false 
      )}
      
      </div>
);

export default ListOfProjects;