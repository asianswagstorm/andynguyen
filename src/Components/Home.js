import React, { Component } from "react";
import "../css/Home.css";
import "../css/font-awesome.min.css";
import ListOfProjects from "./ListOfProjects";
import { Wave } from 'react-animated-text';
import Typist from 'react-typist';
import { popUpNotification,sanitize } from "./constants/HelperFunction/Functions";

const languages = [
  { languages: "All" },
  { languages: "Work Experience" },
  { languages: "JavaScript" },
  { languages: "Python" }
];

const socials = [
  {
    link: "https://www.linkedin.com/in/ðŸ˜„-andy-nguyen-84818b15b/",
    type: "LinkedIn"
  },
  {
    link: "https://github.com/asianswagstorm",
    type: "Github"
  },
  {
    link: "https://www.instagram.com/asianswagstorm/?hl=en",
    type: "Instagram"
  }
];

class Home extends Component {
  constructor(props) {
    super(props);
    this.state = {
      filterOption: "",
      isToggleOn: true,
      loaded:false,
      contactInfo: {name: "", _replyto: "", message: ""},
      hasEmpty: true
    };
    this.submitForm = this.submitForm.bind(this);
  }

  componentDidMount = () => {
    this.state.loaded === false && setTimeout(() => this.setState({loaded: true}) , 1000)
  }

  //use redux props clean this up
//<script type="application/text"> console.log("hello") </script>

  submitForm = (event) => {
    event.preventDefault();
    const form = event.target;
    const data = new FormData(form);
    
    if(!this.state.hasEmpty){
      const xhr = new XMLHttpRequest();
      const {name, _replyto, message} = this.state.contactInfo;
      data.set("name", name);
      data.set("_replyto", _replyto);
      data.set("message", message);

      xhr.open(form.method, form.action);
      xhr.setRequestHeader("Accept", "application/json");
      xhr.onreadystatechange = () => {
        if (xhr.readyState !== XMLHttpRequest.DONE) return;
        if (xhr.status === 200) {
          //form.reset();
          popUpNotification('success', "Thanks for reaching out, I will get back to you shortly." )
        } else {
            popUpNotification('error', "Sorry, there was an error while submitting the form." )
        }
      };
      xhr.send(data);
      this.setState({ hasEmpty: true } );
    }
    
  };

  filter = type => {
    this.setState({ filterOption: type });
  };

  setContactInfo = (event) => {
    const {name,value} = event.target;
    const contactInfo = {...this.state.contactInfo, [name] : sanitize(value)} 
    const hasEmpty = Object.values(contactInfo).some(variable => variable.trim(" ") === "")
    this.setState({ contactInfo, hasEmpty } );
  }

  render = () => {
    //use props from redux.
    const language_buttons = languages.map((x, index) => {
      return (
        <li key={index}>
          <button
            onClick={() => this.filter(x.languages !== "All" ? x.languages : "")}
            data-tag={x.languages}
            className="language_button"
          >
            {x.languages}
          </button>
        </li>
      );
    });

    return (
     <div>
        <section id="main">
          <section id="banner">
            <div className="inner">
              <div className="heyThere">
                <h1 className="intro-andy">
                  <Wave text="Hey! " iterations={this.state.loaded === false ? 1 : 0} /> 
                </h1> {/** redux function set timeout */}
                { this.state.loaded === true && 
                <span>
                <h1 className="intro_name">
                  <Wave text="I'm " iterations={1}/>
                
                  <Typist cursor={{ show: false }}  startDelay={800}>
                    Andy.
                  </Typist>
                </h1>
                </span>
                }
             </div>
            </div>
          </section>

          <header id="header" />
          <section id="galleries">
            <div className="gallery">
              <header>
                <h1>Projects / Experience</h1>
                <ul id="tabs">{language_buttons}</ul>
              </header>

              <ListOfProjects filterOption={this.state.filterOption}/>
            </div>
          </section>

          <section id="contact">
            <div className="social column">
              <h3>About Me</h3>
              <p id="about-me">
                My name is Andy Nguyen, a Montreal native software developer. 
                I am a computer science graduate, who's always on the pursuit of knowledge and self improvement. 
              </p>
              <p id="about-me">
                Software development is one of my life passions, because I find value in turning creative ideas into reality.  
                Finding solutions to client facing / challenging problems sparks joy within me.
                I am seeking to join a dynamic, passionate, fun and energetic team whom can provide mentorship opportunities, to help extend my knowledge in software development to greater lengths.
                While having a blast creating awesome software.
                 Looking forward to working with you.
              </p>
              <p id="about-me">
              Outside of coding, I hold a personal training certification and I compete in the 
               <a
                  href="https://www.fqd-quebec.com/athletes/528"
                  target="_blank"
                  rel="noopener noreferrer"
                > Quebec Powerlifting Federation.</a>
                 My passion for fitness has taught me dedication and confidence that directly translates to everything I do in life. 
                 Other hobbies include being an amature chef, a dog enthusiast and a tech admirer.  
              </p>

              <h3>My Socials</h3>
              <ul className="icons">
                {
                socials.map(social => 
                 <li>
                    <a
                      target="_blank"
                      rel="noopener noreferrer"
                      href= {social.link}
                      className={`icon fa-${social.type.toLowerCase()}`}
                    >
                      <span className="label">{social.type}</span>
                    </a>
                </li>
                )  
              }
              </ul>
            </div>
          
            <div className="column">
              <h3>Get in Touch with Me</h3>
              <form
                onSubmit={this.submitForm}
                action="https://formspree.io/xjvyddlb"
                method="POST"
              >
                <div className="field half first">
                  <label htmlFor="name">Name
                  <input name="name" id="name" type="text" placeholder="Name" onChange={this.setContactInfo}/>
                  </label>
                </div>
                <div className="field half">
                  <label htmlFor="_replyto">Email
                  <input
                    name="_replyto"
                    id="_replyto"
                    type="email"
                    placeholder="example@example.com"
                    onChange={this.setContactInfo}
                  />
                  </label>
                </div>
                <div className="field">
                  <label htmlFor="message">Message
                  <textarea
                    name="message"
                    id="message"
                    rows="6"
                    placeholder="i.e I like Dogs!"
                    onChange={this.setContactInfo}
                  />
                  </label>
                </div>
              
                <ul className="actions">
                  <li>
                    <input
                      value="Send Message"
                      className="button"
                      type="submit"
                      disabled={this.state.hasEmpty}
                    />
                  </li>
                </ul> 
              </form>
            </div>
          </section>
        </section>
        </div>
    );
  };
};

export default Home;
