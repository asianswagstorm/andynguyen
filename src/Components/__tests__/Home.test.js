import React from 'react';
import ReactDOM from 'react-dom';
import { cleanup,render } from '@testing-library/react';
import Home from "../Home.js"
import ListOfProjects from "../ListOfProjects.js";
import Card from "../Card.js";
import {shallow} from "enzyme";
import {tiles} from "../constants/Tiles";

afterEach(cleanup);

describe('HomeRenders', () => {
    describe('render', () => {
        it('home renders without crashing', () => {
          const div = document.createElement('div');
          ReactDOM.render(<Home/>, div);
          ReactDOM.unmountComponentAtNode(div);
        });

        it('should render properly', () => {
          const wrapper =  render(<Home/>);
          expect(wrapper).toMatchSnapshot();
        })
      });
    describe('match number of projects', () => {
      it('should match the number of projects', () => {
        const numberOfProjects = tiles.length;
        const wrapper = shallow(<ListOfProjects filterOption=""/>)
        const actualNumberOfProjects = wrapper.find(Card).length;
        expect(actualNumberOfProjects).toEqual(numberOfProjects);
      })
    })
});
