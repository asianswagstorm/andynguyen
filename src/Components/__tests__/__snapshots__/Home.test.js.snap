// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeRenders render should render properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
        <div>
            <section
                id="main"
            >
                <section
                    id="banner"
                >
                    <div
                        class="inner"
                    >
                        <div
                            class="heyThere"
                        >
                            <h1
                                class="intro-andy"
                            >
                                <div>
                                    <span
                                        style="display: inline-block; width: auto;"
                                    >
                                        <span
                                            style="display: inline-block;"
                                        >
                                            H
                                        </span>
                                    </span>
                                    <span
                                        style="display: inline-block; width: auto;"
                                    >
                                        <span
                                            style="display: inline-block;"
                                        >
                                            e
                                        </span>
                                    </span>
                                    <span
                                        style="display: inline-block; width: auto;"
                                    >
                                        <span
                                            style="display: inline-block;"
                                        >
                                            y
                                        </span>
                                    </span>
                                    <span
                                        style="display: inline-block; width: auto;"
                                    >
                                        <span
                                            style="display: inline-block;"
                                        >
                                            !
                                        </span>
                                    </span>
                                    <span
                                        style="display: inline-block; width: 0.5em;"
                                    >
                                        <span
                                            style="display: inline-block;"
                                        >
                                            
                                        </span>
                                    </span>
                                </div>
                            </h1>
                        </div>
                    </div>
                </section>
                <header
                    id="header"
                />
                <section
                    id="galleries"
                >
                    <div
                        class="gallery"
                    >
                        <header>
                            <h1>
                                Projects / Experience
                            </h1>
                            <ul
                                id="tabs"
                            >
                                <li>
                                    <button
                                        data-tag="All"
                                        class="language_button"
                                    >
                                        All
                                    </button>
                                </li>
                                <li>
                                    <button
                                        data-tag="Work Experience"
                                        class="language_button"
                                    >
                                        Work Experience
                                    </button>
                                </li>
                                <li>
                                    <button
                                        data-tag="JavaScript"
                                        class="language_button"
                                    >
                                        JavaScript
                                    </button>
                                </li>
                                <li>
                                    <button
                                        data-tag="Python"
                                        class="language_button"
                                    >
                                        Python
                                    </button>
                                </li>
                            </ul>
                        </header>
                        <div
                            class="content"
                        >
                            <div
                                class="media"
                            >
                                <a
                                    target="_blank"
                                    rel="noopener noreferrer"
                                    href="https://jamdo-487.herokuapp.com/"
                                >
                                    <img
                                        src="images/thumbs/thumb1.jpg"
                                        alt=""
                                        title="https://jamdo-487.herokuapp.com/"
                                    />
                                </a>
                            </div>
                            <div
                                class="media"
                            >
                                <a
                                    target="_blank"
                                    rel="noopener noreferrer"
                                    href="https://airbnbmock.herokuapp.com/"
                                >
                                    <img
                                        src="images/thumbs/thumb3.jpg"
                                        alt=""
                                        title="https://airbnbmock.herokuapp.com/"
                                    />
                                </a>
                            </div>
                            <div
                                class="media"
                            >
                                <a
                                    target="_self"
                                    rel="noopener noreferrer"
                                    href="/Recipes"
                                >
                                    <img
                                        src="images/thumbs/thumb12.jpg"
                                        alt=""
                                        title="/Recipes"
                                    />
                                </a>
                            </div>
                            <div
                                class="media"
                            >
                                <a
                                    target="_self"
                                    rel="noopener noreferrer"
                                    href="/Pokemon"
                                >
                                    <img
                                        src="images/thumbs/thumb13.jpg"
                                        alt=""
                                        title="/Pokemon"
                                    />
                                </a>
                            </div>
                            <div
                                class="media"
                            >
                                <a
                                    target="_self"
                                    rel="noopener noreferrer"
                                    href="/Spotify"
                                >
                                    <img
                                        src="images/thumbs/thumb14.jpg"
                                        alt=""
                                        title="/Spotify"
                                    />
                                </a>
                            </div>
                            <div
                                class="media"
                            >
                                <a
                                    target="_self"
                                    rel="noopener noreferrer"
                                    href="/TicTacToe"
                                >
                                    <img
                                        src="images/thumbs/thumb6.jpg"
                                        alt=""
                                        title="/TicTacToe"
                                    />
                                </a>
                            </div>
                            <div
                                class="media"
                            >
                                <a
                                    target="_self"
                                    rel="noopener noreferrer"
                                    href="/Connect4"
                                >
                                    <img
                                        src="images/thumbs/thumb8.jpg"
                                        alt=""
                                        title="/Connect4"
                                    />
                                </a>
                            </div>
                            <div
                                class="media"
                            >
                                <a
                                    target="_self"
                                    rel="noopener noreferrer"
                                    href="/Experience1"
                                >
                                    <img
                                        src="images/thumbs/thumb7.jpg"
                                        alt=""
                                        title="/Experience1"
                                    />
                                </a>
                            </div>
                        </div>
                    </div>
                </section>
                <section
                    id="contact"
                >
                    <div
                        class="social column"
                    >
                        <h3>
                            About Me
                        </h3>
                        <p
                            id="about-me"
                        >
                            My name is Andy Nguyen, a Montreal based software developer. A recent Computer Science graduate, but am always learning. I chose this field because finding solutions to problems and developing quality software excites me. I am looking for mentorship opportunities within a diverse team, that can challenge me and help develop my skills as a software developer even further. I look forward to working with you.
                        </p>
                        <p
                            id="about-me"
                        >
                            Outside of coding, I hold a personal training certification and I 
                            <a
                                href="http://www.fqd-quebec.com/lifter/528"
                                target="_blank"
                                rel="noopener noreferrer"
                            >
                                compete in powerlifting
                            </a>
                            within the Quebec Powerlifting Federation. My other interest include cooking, mixology, photography and of course petting dogs.
                        </p>
                        <h3>
                            My Socials
                        </h3>
                        <ul
                            class="icons"
                        >
                            <li>
                                <a
                                    target="_blank"
                                    rel="noopener noreferrer"
                                    href="https://www.linkedin.com/in/andy-nguyen-84818b15b/"
                                    class="icon fa-linkedin"
                                >
                                    <span
                                        class="label"
                                    >
                                        LinkedIn
                                    </span>
                                </a>
                            </li>
                            <li>
                                <a
                                    target="_blank"
                                    rel="noopener noreferrer"
                                    href="https://github.com/asianswagstorm"
                                    class="icon fa-github"
                                >
                                    <span
                                        class="label"
                                    >
                                        Github
                                    </span>
                                </a>
                            </li>
                            <li>
                                <a
                                    target="_blank"
                                    rel="noopener noreferrer"
                                    href="https://www.instagram.com/asianswagstorm/?hl=en"
                                    class="icon fa-instagram"
                                >
                                    <span
                                        class="label"
                                    >
                                        Instagram
                                    </span>
                                </a>
                            </li>
                        </ul>
                    </div>
                    <div
                        class="column"
                    >
                        <h3>
                            Get in Touch with Me
                        </h3>
                        <form
                            action="https://formspree.io/xjvyddlb"
                            method="POST"
                        >
                            <div
                                class="field half first"
                            >
                                <label
                                    for="name"
                                >
                                    Name
                                    <input
                                        name="name"
                                        id="name"
                                        type="text"
                                        placeholder="Name"
                                    />
                                </label>
                            </div>
                            <div
                                class="field half"
                            >
                                <label
                                    for="_replyto"
                                >
                                    Email
                                    <input
                                        name="_replyto"
                                        id="email"
                                        type="email"
                                        placeholder="Email"
                                    />
                                </label>
                            </div>
                            <div
                                class="field"
                            >
                                <label
                                    for="message"
                                >
                                    Message
                                    <textarea
                                        name="message"
                                        id="message"
                                        rows="6"
                                        placeholder="Message"
                                    />
                                </label>
                            </div>
                            <ul
                                class="actions"
                            >
                                <li>
                                    <input
                                        class="button"
                                        type="submit"
                                        value="Send Message"
                                    />
                                </li>
                            </ul>
                        </form>
                    </div>
                </section>
            </section>
        </div>
    </div>
</body>,
  "container": <div>
    <div>
        <section
            id="main"
        >
            <section
                id="banner"
            >
                <div
                    class="inner"
                >
                    <div
                        class="heyThere"
                    >
                        <h1
                            class="intro-andy"
                        >
                            <div>
                                <span
                                    style="display: inline-block; width: auto;"
                                >
                                    <span
                                        style="display: inline-block;"
                                    >
                                        H
                                    </span>
                                </span>
                                <span
                                    style="display: inline-block; width: auto;"
                                >
                                    <span
                                        style="display: inline-block;"
                                    >
                                        e
                                    </span>
                                </span>
                                <span
                                    style="display: inline-block; width: auto;"
                                >
                                    <span
                                        style="display: inline-block;"
                                    >
                                        y
                                    </span>
                                </span>
                                <span
                                    style="display: inline-block; width: auto;"
                                >
                                    <span
                                        style="display: inline-block;"
                                    >
                                        !
                                    </span>
                                </span>
                                <span
                                    style="display: inline-block; width: 0.5em;"
                                >
                                    <span
                                        style="display: inline-block;"
                                    >
                                        
                                    </span>
                                </span>
                            </div>
                        </h1>
                    </div>
                </div>
            </section>
            <header
                id="header"
            />
            <section
                id="galleries"
            >
                <div
                    class="gallery"
                >
                    <header>
                        <h1>
                            Projects / Experience
                        </h1>
                        <ul
                            id="tabs"
                        >
                            <li>
                                <button
                                    data-tag="All"
                                    class="language_button"
                                >
                                    All
                                </button>
                            </li>
                            <li>
                                <button
                                    data-tag="Work Experience"
                                    class="language_button"
                                >
                                    Work Experience
                                </button>
                            </li>
                            <li>
                                <button
                                    data-tag="JavaScript"
                                    class="language_button"
                                >
                                    JavaScript
                                </button>
                            </li>
                            <li>
                                <button
                                    data-tag="Python"
                                    class="language_button"
                                >
                                    Python
                                </button>
                            </li>
                        </ul>
                    </header>
                    <div
                        class="content"
                    >
                        <div
                            class="media"
                        >
                            <a
                                target="_blank"
                                rel="noopener noreferrer"
                                href="https://jamdo-487.herokuapp.com/"
                            >
                                <img
                                    src="images/thumbs/thumb1.jpg"
                                    alt=""
                                    title="https://jamdo-487.herokuapp.com/"
                                />
                            </a>
                        </div>
                        <div
                            class="media"
                        >
                            <a
                                target="_blank"
                                rel="noopener noreferrer"
                                href="https://airbnbmock.herokuapp.com/"
                            >
                                <img
                                    src="images/thumbs/thumb3.jpg"
                                    alt=""
                                    title="https://airbnbmock.herokuapp.com/"
                                />
                            </a>
                        </div>
                        <div
                            class="media"
                        >
                            <a
                                target="_self"
                                rel="noopener noreferrer"
                                href="/Recipes"
                            >
                                <img
                                    src="images/thumbs/thumb12.jpg"
                                    alt=""
                                    title="/Recipes"
                                />
                            </a>
                        </div>
                        <div
                            class="media"
                        >
                            <a
                                target="_self"
                                rel="noopener noreferrer"
                                href="/Pokemon"
                            >
                                <img
                                    src="images/thumbs/thumb13.jpg"
                                    alt=""
                                    title="/Pokemon"
                                />
                            </a>
                        </div>
                        <div
                            class="media"
                        >
                            <a
                                target="_self"
                                rel="noopener noreferrer"
                                href="/Spotify"
                            >
                                <img
                                    src="images/thumbs/thumb14.jpg"
                                    alt=""
                                    title="/Spotify"
                                />
                            </a>
                        </div>
                        <div
                            class="media"
                        >
                            <a
                                target="_self"
                                rel="noopener noreferrer"
                                href="/TicTacToe"
                            >
                                <img
                                    src="images/thumbs/thumb6.jpg"
                                    alt=""
                                    title="/TicTacToe"
                                />
                            </a>
                        </div>
                        <div
                            class="media"
                        >
                            <a
                                target="_self"
                                rel="noopener noreferrer"
                                href="/Connect4"
                            >
                                <img
                                    src="images/thumbs/thumb8.jpg"
                                    alt=""
                                    title="/Connect4"
                                />
                            </a>
                        </div>
                        <div
                            class="media"
                        >
                            <a
                                target="_self"
                                rel="noopener noreferrer"
                                href="/Experience1"
                            >
                                <img
                                    src="images/thumbs/thumb7.jpg"
                                    alt=""
                                    title="/Experience1"
                                />
                            </a>
                        </div>
                    </div>
                </div>
            </section>
            <section
                id="contact"
            >
                <div
                    class="social column"
                >
                    <h3>
                        About Me
                    </h3>
                    <p
                        id="about-me"
                    >
                        My name is Andy Nguyen, a Montreal based software developer. A recent Computer Science graduate, but am always learning. I chose this field because finding solutions to problems and developing quality software excites me. I am looking for mentorship opportunities within a diverse team, that can challenge me and help develop my skills as a software developer even further. I look forward to working with you.
                    </p>
                    <p
                        id="about-me"
                    >
                        Outside of coding, I hold a personal training certification and I 
                        <a
                            href="http://www.fqd-quebec.com/lifter/528"
                            target="_blank"
                            rel="noopener noreferrer"
                        >
                            compete in powerlifting
                        </a>
                        within the Quebec Powerlifting Federation. My other interest include cooking, mixology, photography and of course petting dogs.
                    </p>
                    <h3>
                        My Socials
                    </h3>
                    <ul
                        class="icons"
                    >
                        <li>
                            <a
                                target="_blank"
                                rel="noopener noreferrer"
                                href="https://www.linkedin.com/in/andy-nguyen-84818b15b/"
                                class="icon fa-linkedin"
                            >
                                <span
                                    class="label"
                                >
                                    LinkedIn
                                </span>
                            </a>
                        </li>
                        <li>
                            <a
                                target="_blank"
                                rel="noopener noreferrer"
                                href="https://github.com/asianswagstorm"
                                class="icon fa-github"
                            >
                                <span
                                    class="label"
                                >
                                    Github
                                </span>
                            </a>
                        </li>
                        <li>
                            <a
                                target="_blank"
                                rel="noopener noreferrer"
                                href="https://www.instagram.com/asianswagstorm/?hl=en"
                                class="icon fa-instagram"
                            >
                                <span
                                    class="label"
                                >
                                    Instagram
                                </span>
                            </a>
                        </li>
                    </ul>
                </div>
                <div
                    class="column"
                >
                    <h3>
                        Get in Touch with Me
                    </h3>
                    <form
                        action="https://formspree.io/xjvyddlb"
                        method="POST"
                    >
                        <div
                            class="field half first"
                        >
                            <label
                                for="name"
                            >
                                Name
                                <input
                                    name="name"
                                    id="name"
                                    type="text"
                                    placeholder="Name"
                                />
                            </label>
                        </div>
                        <div
                            class="field half"
                        >
                            <label
                                for="_replyto"
                            >
                                Email
                                <input
                                    name="_replyto"
                                    id="email"
                                    type="email"
                                    placeholder="Email"
                                />
                            </label>
                        </div>
                        <div
                            class="field"
                        >
                            <label
                                for="message"
                            >
                                Message
                                <textarea
                                    name="message"
                                    id="message"
                                    rows="6"
                                    placeholder="Message"
                                />
                            </label>
                        </div>
                        <ul
                            class="actions"
                        >
                            <li>
                                <input
                                    class="button"
                                    type="submit"
                                    value="Send Message"
                                />
                            </li>
                        </ul>
                    </form>
                </div>
            </section>
        </section>
    </div>
</div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
